jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java

```diff
@@ -334,7 +335,7 @@ public CachedHttpContent(String key, HttpContent httpContent)
                 throw new IllegalArgumentException("Resource is too large: length " + contentLengthValue + " > " + _maxCachedFileSize);
 
             // Read the content into memory
-            _buffer = IOResources.toRetainableByteBuffer(httpContent.getResource(), _bufferPool);
+            _buffer.set(IOResources.toRetainableByteBuffer(httpContent.getResource(), _bufferPool));
```

Comment:
Can `IOResources.toRetainableByteBuffer()` ever return null?

Reply:
The null check is needed because we now null the reference when the buffer is released